src/libraries/System.Private.Xml/src/System/Xml/Core/XmlEncodedRawTextWriterAsync.cs

```diff
@@ -73,6 +73,63 @@ internal override Task WriteXmlDeclarationAsync(string xmldecl)
             return Task.CompletedTask;
         }
 
+        protected override async ValueTask DisposeAsyncCore()
+        {
+            try
+            {
+                await FlushBufferAsync().ConfigureAwait(false);
+            }
+            finally
+            {
+                // Future calls to Close or Flush shouldn't write to Stream or Writer
+                writeToNull = true;
+
+                if (stream != null)
```

Comment:
Nit: Not related to this change, but it's confusing to me to see fields without the underscore prefix.  It'd be nice to clean that up in the future.

Reply:
Created an issue for tracking https://github.com/dotnet/runtime/issues/35320